### DISCUSSION TITLE
How to set up the default session when embedding JB in my web app

### DISCUSSION THREAD

**Discussion by ygao61** [GMOD/jbrowse-components#4093] [Q&A]
Hi, 

[Closed.]

I figured it out and it works now. the 'id' doesn't matter and make sure the tracks-> configuration is matching your trackId in assembly.js. The configuration for display share the same prefix of it.

_____________________________________

I am new to the web development and JBrowse. I've met some issues about the index.html, thanks a lot in advance for your help.

I wanted a **default session** for my data(1 assembly and multiple tracks). I succeed with showing 'the default session' in tutorial. 
I am not sure what to pass to the 'defaultSession' when creating a state. (I didn't find the answer in the tutorial page, lmk if I missed anything)
Like, the line 54 in the attached img. I pass the configuration: 'farr1-ReferenceSequenceTrack', which match my trackId in my assembly.js

**I don't know how to change the other part. For instance, what should the 'tracks{ id:} 'be? and what should I pass to the 'display{id, type and configuration}'**

![Screenshot 2023-11-27 at 10 42 00 PM](https://github.com/GMOD/jbrowse-components/assets/107954058/15d68f2e-20a9-43e2-8ecf-c53682270694)

Thanks,
Yuan

**Comment by cmdcolin** [comment]
hi there, glad you figured this out. 

it is true that the defaultSession is a little hard to handcode, because it is sort of a snapshot of the state of the app and contains a lot of things that aren't exactly the most intuitive.

as an alternative, you can actually manually call functions on the value returned by createViewState

an example of this is here, I manually call showTrack with the trackId, and then that is shown
https://github.com/GMOD/jbrowse-components/blob/main/products/jbrowse-react-linear-genome-view/stories/examples/WithShowTrack.tsx

that way, you don't have to deal with all the intricacies of the defaultSession format

**Reply by cmdcolin** [reply]
(example above came from our storybook https://jbrowse.org/storybook/lgv/main/)

**Reply by ygao61** [reply]
Thanks, the showTrack seems pretty cool!
